tests/span_compatibility_tests.cpp

```diff
@@ -42,11 +42,51 @@ static_assert(std::is_convertible<Derived*, Base*>::value, "std::is_convertible<
 static_assert(!std::is_convertible<Derived (*)[], Base (*)[]>::value,
               "!std::is_convertible<Derived(*)[], Base(*)[]>");
 
+// int*(*) [], int const* const(*)[] was identified as an issue in CWG330 and the resolution was
+// provided with N4261. The changes were not backported to all versions of the compilers that GSL
+// supports. The `if constexpr` should prevent codegen from happening if it is not supported however
+// a few compilers continue to complain about the logic within. Filtering g++ version < 8, clang
+// version < 7, and XCode 9.4 and 10.1 which looks to be Apple clang versions 9.1 and 10.0.
+#if (defined(_MSC_VER)) || (defined(__GNUC__) && __GNUC__ > 7) ||                                  \
+    (defined(__APPLE__) && defined(__clang_major__) && __clang_major__ > 10) ||                    \
+    (!defined(__APPLE__) && defined(__clang__) && __clang_major__ > 6)
+template <class = void>
```

Comment:
If you change this template parameter to `class T = int`, and replace the `int`s with `T`s in the function body, it will probably fix most of the stuff you're avoiding with the messy conditional on lines 50-52.

Reply:
Thanks @CaseyCarter, that did the trick. I'll merge this tomorrow morning, since Appveyor is still running.